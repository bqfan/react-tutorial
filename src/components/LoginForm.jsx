import { useEffect } from "react"
import { useDocumentClick } from "../utils/hooks/useDocumnetClick";

export function LoginForm() {
    const resizeEventHandler = (e) => {
        console.log("Window/ViewPort Resized!");
    }

    useEffect(() => {
        window.addEventListener('resize', resizeEventHandler);

        return () => {
            console.log("Unmounting LoginForm");
            console.log("Removing resize eventListener")
            window.removeEventListener('resize', resizeEventHandler);
        };
    }, []);

    useDocumentClick();

    return <form onSubmit={(e) => {
        e.preventDefault();
        const formData = new FormData(e.target);
        const username = formData.get("username");
        const password = formData.get("password");

        // fetch("http://localhost:3001/api/login", {
        //     body: {
        //         username,
        //         password,
        //     },
        // });
    }}>
        <label htmlFor="username">Username</label>
        <br />
        <input
            id="username"
            type="text"
            name="username"
            autoComplete="username"
            required
            onChange={(e) => {
                console.log(`Username: ${e.target.value}`);
            }}
        />
        <br />
        <label htmlFor="password">Password</label>
        <br />
        <input
            id="password"
            type="password"
            name="password"
            autoComplete="current-password"
            required
            onChange={(e) => {
                console.log(`Password: ${e.target.value}`);
            }}
        />
        <br />
        <button type="submit">Login</button>
    </form>
}
