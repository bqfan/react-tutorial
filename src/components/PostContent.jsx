import { useContext } from "react";
import { PostContentButtons } from "./PostContentButtons";
import { UserContext } from "../utils/contexts/UserContext";

export function PostContent() {
    const userContextData = useContext(UserContext);

    return (
        <div>
            <div>
                <span>PostContent</span>
            </div>
            <PostContentButtons />
            {userContextData.id}
            {userContextData.displayName}
            {userContextData.email}
            {userContextData.username}
        </div>
    )
}