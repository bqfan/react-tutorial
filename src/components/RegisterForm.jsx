import { useState } from "react";

export function RegisterForm() {
    // const [username, setUsername] = useState("");
    // const [password, setPassword] = useState("");
    // const [displayName, setDisplayName] = useState("");

    // const isDisabled = !username || !password || !displayName;

    const [formFields, setFormFields] = useState({
        username: "",
        password: "",
        displayName: "",
    });

    const isDisabled = !formFields.username || !formFields.password || !formFields.displayName;

    console.log(formFields);

    return <form>
        <div>
            <label htmlFor="username">Username: </label>
            <br />
            <input
                id="username"
                type="text"
                name="username"
                value={formFields.username}
                autoComplete="username"
                required 
                // onChange={(e) => setUsername(e.target.value)}
                onChange={(e) => {
                    setFormFields((currentState) =>
                        ({
                            ...currentState,
                            username: e.target.value
                        }));
                }}
            />
            <br />
        </div>
        <div>
            <label htmlFor="password">Password: </label>
            <br />
            <input
                id="password"
                type="text"
                name="password"
                value={formFields.password}
                autoComplete="new-password"
                required
                // onChange={(e) => setPassword(e.target.value)}
                onChange={(e) => {
                    setFormFields((currentState) =>
                        ({
                            ...currentState,
                            password: e.target.value
                        }));
                }}
            />
            <br />
        </div>
        <div>
            <label htmlFor="displayName">Display name: </label>
            <br />
            <input
                id="displayName"
                type="text"
                name="displayName"
                value={formFields.displayName}
                autoComplete="displayName"
                required
                // onChange={(e) => setDisplayName(e.target.value)}
                onChange={(e) => {
                    setFormFields((currentState) =>
                        ({
                            ...currentState,
                            displayName: e.target.value
                        }));
                }}
            />
            <br />
        </div>
        <div>
            <button disabled={isDisabled}>Sign up</button>
        </div>
        <div>
            <span>Username: {formFields.username}</span>
        </div>
        <div>
            <span>Password: {formFields.password}</span>
        </div>
        <div>
            <span>Display Name: {formFields.displayName}</span>
        </div>
    </form>
}