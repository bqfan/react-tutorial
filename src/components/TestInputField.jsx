export function TestInputField() {
    return <input placeholder="Enter data..." value="hello" />;
}
