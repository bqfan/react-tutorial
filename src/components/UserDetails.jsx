import PropTypes from 'prop-types'
import { useState} from "react"

export function UserDetails(props) {
    const [isEditing, setIsEditing] = useState(false);
    const [username, setUsername] = useState(props.user.username)
    const [email, setEmail] = useState(props.user.email)
    // console.log(isEditing);
    return (
        <div>
            <div>
                <button onClick={() => {
                    setIsEditing((currentState) => !currentState);
                }}>Edit</button>
                <button>Delete</button>
                <button onClick={() => console.log(props.user)}>Save</button>
            </div>
            <div key={props.user.id}>
                <b>ID: </b>
                <span>{props.user.id}</span>
                <br />
                <b>Username: </b>
                {isEditing ? <input
                    id="username"
                    type="text"
                    name="username"
                    value={username}
                    autoComplete="username"
                    required
                    onChange={(e) => {
                        setUsername(e.target.value);
                    }} /> :
                    <span>{props.user.username}</span>}
                <br />
                <b>Email: </b>
                {isEditing ? <input
                    id="email"
                    type="email"
                    name="email"
                    value={email}
                    autoComplete="email"
                    required
                    onChange={(e) => {
                        setEmail(e.target.value);
                    }} /> :
                    <span>{props.user.email}</span>}
                <br />
            </div>
        </div>
    )
}

UserDetails.propTypes = {
    user: PropTypes.shape({
        id: PropTypes.number.isRequired,
        username: PropTypes.string.isRequired,
        email: PropTypes.string.isRequired,
    }).isRequired,
    setUsers: PropTypes.func.isRequired,
}