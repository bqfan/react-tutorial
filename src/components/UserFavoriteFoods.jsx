import { createElement } from 'react';
import styles from "./users.module.scss"

export function UserFavoriteFoods() {
    return createElement('section', null,
        <span className={styles["foods-title"]}>Favorite Foods:</span>,
        <br />,
        <ul>
            <li>Sushi</li>
            <li>Pizza</li>
            <li>mediterranean Food</li>
        </ul>
    )
}
