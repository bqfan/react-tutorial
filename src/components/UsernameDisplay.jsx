export function UsernameDisplay({ username }) {
    return (
        <div>
            {/* <span>{username}</span> */}
            <span></span>
        </div>
    );
}
