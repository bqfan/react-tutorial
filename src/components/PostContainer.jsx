import { useContext, useState } from "react"
import { PostContent } from "./PostContent";
import { UserContext } from "../utils/contexts/UserContext";

export function PostContainer() {
    const [data, setData] = useState("Hello World!");
    const userContextData = useContext(UserContext);

    console.log(userContextData);
    return (
        <div>
            <div>
                <h3>PostContainer</h3>
            </div>
            <div>Display Name: {userContextData.name}</div>
            <div>ID: {userContextData.id}</div>
            <div>Email: {userContextData.email}</div>
            <div>Username: {userContextData.username}</div>
            <PostContent />
        </div>
    )
}
