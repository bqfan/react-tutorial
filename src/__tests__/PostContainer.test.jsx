import { expect, it, describe } from 'vitest';
import { render, screen } from "@testing-library/react";
import { PostContainer } from '../components/PostContainer';
import { UserContext } from '../utils/contexts/UserContext';

describe('render context values', () => {
    it('should match snapshot', () => {
        const mockUserContextData = {
            id: 1001,
            username: "user1001",
            email: "user1001@example.com",
            displayName: "User1001",
            setUserData: () => {},
        };

        const { container } = render(
            <UserContext.Provider value={mockUserContextData}>
                <PostContainer />;
            </UserContext.Provider>
        );

        expect(container).toMatchSnapshot();
    })

    it('should display correct username', () => {
        const mockUserContextData = {
            id: 1001,
            username: "user1001",
            email: "user1001@example.com",
            displayName: "User1001",
            setUserData: () => {},
        };

        const { container } = render(
            <UserContext.Provider value={mockUserContextData}>
                <PostContainer />;
            </UserContext.Provider>
        );

        expect(screen.getByText('Username: user1001')).toBeInTheDocument();

    })
})
