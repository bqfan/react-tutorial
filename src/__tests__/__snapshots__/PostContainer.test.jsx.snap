// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`render context values > should match snapshot 1`] = `
<div>
  <div>
    <div>
      <h3>
        PostContainer
      </h3>
    </div>
    <div>
      Display Name: 
    </div>
    <div>
      ID: 
      1001
    </div>
    <div>
      Email: 
      user1001@example.com
    </div>
    <div>
      Username: 
      user1001
    </div>
    <div>
      <div>
        <span>
          PostContent
        </span>
      </div>
      <div>
        <div>
          <h3>
            PostContentButtons
          </h3>
        </div>
        1001
        <br />
        <label
          for="updateName"
        >
          Update Name: 
        </label>
        <input
          id="updateName"
          value=""
        />
        <button>
          Update DisplayName
        </button>
      </div>
      1001
      User1001
      user1001@example.com
      user1001
    </div>
  </div>
  ;
</div>
`;
